### ISSUE TITLE
Monitoring salt event bus crashes

### ISSUE THREAD

**Issue by mchugh19** [carlpett/salt-vault#10]
Just attempted to setup a salt-master, but got bad request response. 
```
[ERROR   ] Failed to get token from master! An error was returned: Bad Request
[ERROR   ] Failed to read secret! CommandExecutionError: {'error': 'Bad Request'}
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/cli/caller.py", line 197, in call
    ret['return'] = func(*args, **kwargs)
  File "/var/cache/salt/minion/extmods/modules/vault.py", line 48, in read_secret
    raise salt.exceptions.CommandExecutionError(e)
CommandExecutionError: {'error': 'Bad Request'}
Error running 'vault.read_secret': {'error': 'Bad Request'}
```

In trying to get to the root of the problem, I looked at the salt event bus, but doing so generated utf8 errors. 
```
Exception occurred in runner state.event: Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/client/mixins.py", line 395, in _low
    data['return'] = self.functions[fun](*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/salt/runners/state.py", line 205, in event
    node=node)
  File "/usr/lib/python2.7/site-packages/salt/modules/state.py", line 1768, in event
    indent=None if not pretty else 4)))
  File "/usr/lib64/python2.7/json/__init__.py", line 251, in dumps
    sort_keys=sort_keys, **kw).encode(obj)
  File "/usr/lib64/python2.7/json/encoder.py", line 209, in encode
    chunks = list(chunks)
  File "/usr/lib64/python2.7/json/encoder.py", line 434, in _iterencode
    for chunk in _iterencode_dict(o, _current_indent_level):
  File "/usr/lib64/python2.7/json/encoder.py", line 408, in _iterencode_dict
    for chunk in chunks:
  File "/usr/lib64/python2.7/json/encoder.py", line 313, in _iterencode_list
    yield buf + _encoder(value)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xf3 in position 1: invalid continuation byte
```

I added a print line to python's json file to see what it was choking on to see:
```
{'fun': 'runner.vault.generate_token', 'fun_args': ['ip-10-241-208-27.ec2.internal', '\x7f\xf3=\xf2\xdf\xf6\xae*\xc8 B\xe9\xa0_\x11\xd6\x12C\xc6\x14S\x00\x89\xfa\xcb\xe5\x8d\xffRxS\xb9 \xd43\xda?S="\xc9\xe1\xa7\xe8\xc9x\xcfK\xf5\xd7\xbc?h\xfd\xe3\\1\x81&\xc4b\x80\x88\x1c\xdfq\xfba\x0f%\xed\x9bV_\x84s\x05.O\x89\xd3i\xa6>\x06\x9e$\xf2\xf6Sx\xf9>\'\xc1&z\xc6w\xa7D\x04O\xf5\x9c~\xdfv\xb8\x11_.@\xb3\xbf\xdcj\xf3:\xc5x\xec\xbdga!/\x90\x03\t\xf8\x17\x13j\xe3\x07\xe5\xce\x95\x8e\xcd\xbb\xc6\x93d\x93\x9dv\xff\x17\xb2\xae\x11\x8e\xban\xf8\xd0\x1d\x18\x13W\xaf\tA\x98\xbb\xdc\xe7v\'\x9au\xd2J\xfb\x02`\x80\xd35\xddp\x953A\x98\xfa\xb2S\x05\xd3\x1b\xa7\xaf\xb3[P\xec\x98f^\xfb4~wHw\x19\xf0\xf4\xf6\xdbk\xd4\x95\xad:$\xaf\x11\x19\x81(\xa6(7#\xe5g\xbc\r\xc8\xd8\xa2k*\xdb\xe5\x13\x1a\xb2c\x12\xefW\x82\xbd\xff4(\xe7\x1d\x1f7\xdc'], 'jid': '20170124220335800898', 'user': 'root', '_stamp': '2017-01-24T22:03:36.316126'}
```

**Comment by carlpett** [comment]
Hi,
Thanks for reporting this, and very sorry for not responding! For some reason I hadn't set Github to watch the repo, so I didn't get notified...
Bit surprised that we haven't run into this - does it happen every time in your setup?

**Comment by mchugh19** [comment]
This can be closed with the merge of #11
